activerecord/lib/active_record/connection_adapters/sqlite3/database_statements.rb

```diff
@@ -34,21 +34,11 @@ def begin_db_transaction # :nodoc:
         end
 
         def commit_db_transaction # :nodoc:
-          log("commit transaction", "TRANSACTION") do
-            with_raw_connection(allow_retry: true, materialize_transactions: false) do |conn|
-              conn.commit
-            end
-          end
-          reset_read_uncommitted
+          internal_execute("COMMIT TRANSACTION", "TRANSACTION", allow_retry: true, materialize_transactions: false)
```

Comment:
`allow_retry: true, materialize_transactions: false` is a bit weird. MySQL and Postgres do the opposite.

But I chose not to change that here, trying to stay focused.